src/github.ts

```diff
@@ -520,6 +536,72 @@ export class GitHub {
     };
   }
 
+  private async *mergeCommitIteratorREST(
+    targetBranch: string,
+    options: CommitIteratorOptions = {}
+  ): AsyncGenerator<Commit> {
+    const maxResults = options.maxResults ?? Number.MAX_SAFE_INTEGER;
+    let results = 0;
+
+    const iterator = this.octokit.paginate.iterator(
+      this.octokit.rest.repos.listCommits,
+      {
+        owner: this.repository.owner,
+        repo: this.repository.repo,
+        per_page: 25,
+        sha: targetBranch,
+      }
+    );
+
+    for await (const response of iterator) {
+      for (let i = 0; i < response.data.length; i++) {
+        if ((results += 1) > maxResults) {
+          break;
+        }
+
+        const commitData = response.data[i];
+        const commit: Commit = {
+          sha: commitData.sha,
+          message: commitData.commit.message,
+          files: await this.getCommitFiles(commitData.sha),
+        };
+
+        const associatedPRs =
+          await this.octokit.repos.listPullRequestsAssociatedWithCommit({
+            owner: this.repository.owner,
+            repo: this.repository.repo,
+            commit_sha: commit.sha,
+          });
+        if (associatedPRs.data.length) {
+          const pullRequest = associatedPRs.data.find(
+            pr => pr.merge_commit_sha === commit.sha
+          );
+          if (pullRequest) {
+            commit.pullRequest = {
+              sha: commit.sha,
+              number: pullRequest.number,
+              baseBranchName: pullRequest.base.ref,
+              headBranchName: pullRequest.head.ref,
+              title: pullRequest.title,
+              body: pullRequest.body ?? '',
+              labels: pullRequest.labels.map(label => label.name),
+              files: commit.files ?? [],
+            };
+          } else {
+            this.logger.warn(
+              `Found ${associatedPRs.data.length} PRs associated with ${commit.sha} but none matched the commit SHA.`
+            );
+          }
+        }
+
+        yield commit;
```

Comment:
Would probably be mildly nice to extract out a function to marshall all this commit data from a given `response.data[i]` but I don't feel strongly, really it's just if we want to submit upstream.